src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EmailAddressAttribute.cs

```diff
@@ -30,20 +30,12 @@ public override bool IsValid(object value)
 
             // only return true if there is only 1 '@' character
             // and it is neither the first nor the last character
-            bool found = false;
-            for (int i = 0; i < valueAsString.Length; i++)
-            {
-                if (valueAsString[i] == '@')
-                {
-                    if (found || i == 0 || i == valueAsString.Length - 1)
-                    {
-                        return false;
-                    }
-                    found = true;
-                }
-            }
-
-            return found;
+            int index = valueAsString.IndexOf('@');
+            
+            return
+                index > 0 &&
+                index != valueAsString.Length - 1 &&
+                index == valueAsString.LastIndexOf('@');
```

Comment:
```suggestion
                0 == valueAsString.AsSpan(index).LastIndexOf('@');
```
Maybe this would be a little more performant on large strings

Reply:
Or actually it probably just iterates backwards